CONTRIBUTING.rst

```diff
@@ -63,6 +63,23 @@ Once that is done, we suggest you:
 General Tips
 ============
 
+Meaningful Commit Messages
+--------------------------
+
+Please choose a meaningful commit message. The commit message is not only
+valuable during the review process, but can be helpful for reasoning about
+any changes in the code base. For example, IntelliJ's "Annotate" feature,
+brings up the commits which introduced the code in a source file. Without
+meaningful commit messages, the commit history does not provide any valuable
+information.
+
+The subject of the commit message (i.e. first line) should contain a summary
+of the changes. It can be prefixed with "Test: " or "Docs: " to indicate
```

Comment:
can we specify here that the first line should read like a command to do something (i.e. not past tense)

for example, this is good:

"Fix flakiness of JoinIntegrationTest"

this is bad:

"Fixed flakiness of JoinIntegrationTest"

this is also bad:

"Fixes flakiness of JoinIntegrationTest"

imagine the text as the label on a button that you press to apply the change. does it make sense?

Reply:
Not sure if we want to enforce imperative style commit subjects. I prefer them as well but there is not a huge difference compared to present or past tense.

Maybe other could chime in if they would like this as well?

Reply:
I have a preference on the present tense for the commits, but not strong opinion.

I think our main goal should be to just be consistent from now on with what we choose and not have mixed style commit msgs.

Reply:
The imperative has the advantage of using less space. Also it is the Linux kernel model and they are pretty much the reference for Git usage :)

The tense doesn't add meaning to the commit message. It's just style, writing meaningful commit messages was what I wanted to point out here. Unless everybody agrees on the tense, I wouldn't change anything in this regard.

Reply:
+1 for imperative form.

Reply:
Looks like there is a preference for formalizing it and to use imperative tense. Will push a fixup.